packages/@aws-cdk/aws-lambda/lib/lambda.ts

```diff
@@ -89,6 +90,31 @@ export interface FunctionProps {
      * Both supplied and generated roles can always be changed by calling `addToRolePolicy`.
      */
     role?: iam.Role;
+
+    /**
+     * VPC network to place Lambda network interfaces
+     *
+     * Specify this if the Lambda function needs to access resources in a VPC.
+     */
+    vpc?: ec2.VpcNetworkRef;
+
+    /**
+     * Where to place the network interfaces within the VPC.
+     *
+     * Only used if 'vpc' is supplied.
+     *
+     * @default Private subnets
+     */
+    vpcPlacement?: ec2.VpcPlacementStrategy;
```

Comment:
Why would that be? I mean, it might be true, but since it's just an ENI I'm guessing you can place it anywhere.

We also still have isolated subnets.

Reply:
This blog post seems to imply you need to do something different for public and private subnets, thereby implying it could be in a public subnet: https://aws.amazon.com/premiumsupport/knowledge-center/internet-access-lambda-function/

Reply:
Thanks for the clarification @heitorlessa. I guess I was wrong.

That still leaves the opportunity to have `Private` (NAT routed) and `Isolated` (nonrouted) subnets, so we still need a placement argument, but now complicated by the fact that selecting public subnets is an error, and we might not know a priori what the selection is going to match.

For the moment, I'll propose to fix this with docs. Everyone okay with this?

Reply:
Fixed it with a post-selection check.